src/lib/browser/Page.ts

```diff
@@ -144,7 +144,14 @@ export class BrowserPage {
 
     await this.#page!.evaluate(injectBaseHref, baseHref);
     return (await this.#page!.evaluate(
-      'document.firstElementChild.outerHTML'
+      () => {
+        const rootTag = document.documentElement.tagName.toLowerCase();
+        const rootAttr = [...document.documentElement.attributes]
+          .map(({ name, value }) => `${name}="${value}"`)
+          .join(' ');
+        const innerContent = (document.documentElement as any).getInnerHTML();
```

Comment:
Shouldn't it needs to `{includeShadowRoots: true}`?

Reply:
The open mode ShadowDOM can be obtained without passing this parameter.

> In order to preserve encapsulation semantics, any [closed shadow roots](https://developer.mozilla.org/docs/Web/API/ShadowRoot/mode) within an element will not be serialized by default.

The default behavior seems to be what we want